hub/models.py

```diff
@@ -1325,12 +1337,20 @@ def get_scheduled_batch_job_progress(self, parent_job: ProcrastinateJob, user=No
         jobs = self.event_log_queryset().filter(args__request_id=request_id).all()
         status = "todo"
 
+        number_of_jobs_ahead_in_queue = (
+            ProcrastinateJob.objects.filter(id__lt=parent_job.id)
+            .filter(status__in=["todo", "doing"])
+            .count()
+        )
+
         if any([job.status == "doing" for job in jobs]):
             status = "doing"
         elif any([job.status == "failed" for job in jobs]):
             status = "failed"
         elif all([job.status == "succeeded" for job in jobs]):
             status = "succeeded"
+        elif number_of_jobs_ahead_in_queue <= 0:
+            status = "succeeded"
```

Comment:
question: what's the logic here?

Reply:
The GraphQL query was responding with status succeeded even if you had `cancelled` some of the jobs. So I added this logic: if there are no more jobs, report that the status is done. There wasn't a `done` status, so I settled for `succeeded` because I'm an optimist.